inc/media.php

```diff
@@ -1058,11 +1058,18 @@ function media_preview_buttons($image, $auth, $rev='') {
  * @return array|false
  */
 function media_image_preview_size($image, $rev, $meta, $size = 500) {
-    if (!preg_match("/\.(jpe?g|gif|png)$/", $image) || !file_exists(mediaFN($image, $rev))) return false;
-
-    $info = getimagesize(mediaFN($image, $rev));
-    $w = (int) $info[0];
-    $h = (int) $info[1];
+    if (!file_exists(mediaFN($image, $rev))) return false;
+ 
+    if (preg_match("/\.svg$/", $image) ) {
+        $xml = simplexml_load_file(mediaFN($image, $rev));
+        $attr = $xml->attributes();
+        $w = (int) $attr->width ;
+        $h = (int) $attr->height ;
```

Comment:
The problem here is that width and height could contain arbitrary SVG units. Eg. 105cm or 100% so simply casting them to integer would result in something wrong.

We could argue it's better than nothing though...